interpreter/cling/lib/Utils/AST.cpp

```diff
@@ -1441,14 +1457,26 @@ namespace utils {
     LookupResult R(*S, Name, SourceLocation(), Sema::LookupOrdinaryName,
                    Sema::ForRedeclaration);
     Lookup::Named(S, R, Within);
-    if (R.empty())
-      return 0;
+    return LookupResult2Decl<clang::NamedDecl>(R);
+  }
 
-    R.resolveKind();
+  TagDecl* Lookup::Tag(Sema* S, llvm::StringRef Name,
+                           const DeclContext* Within) {
+    DeclarationName DName = &S->Context.Idents.get(Name);
+    return Lookup::Tag(S, DName, Within);
+  }
 
-    if (R.isSingleResult())
-      return R.getFoundDecl();
-    return (clang::NamedDecl*)-1;
+  TagDecl* Lookup::Tag(Sema* S, const char* Name,
+                           const DeclContext* Within) {
+    return Lookup::Tag(S, llvm::StringRef(Name), Within);
+  }
+
+  TagDecl* Lookup::Tag(Sema* S, const clang::DeclarationName& Name,
+               const DeclContext* Within) {
+    LookupResult R(*S, Name, SourceLocation(), Sema::LookupTagName,
+                   Sema::ForRedeclaration);
+    Lookup::Named(S, R, Within);
+    return LookupResult2Decl<clang::TagDecl>(R);
```

Comment:
The clang::LookupResult supports filters. They can likely help and you could remove or simplify that routine. We use filters somewhere in TCling, else just grep for LookupFilter.

Reply:
`git grep LookupFilter -- core/metacling/src/` is empty. Also I'm not sure what to filter on here? You mean to only find `TagDecl`s?